controllers/storagecluster/cephobjectstores.go

```diff
@@ -123,6 +125,12 @@ func (obj *ocsCephObjectStores) ensureDeleted(r *StorageClusterReconciler, sc *o
 
 // createCephObjectStore creates CephObjectStore in the desired state
 func (r *StorageClusterReconciler) createCephObjectStores(cephObjectStores []*cephv1.CephObjectStore, instance *ocsv1.StorageCluster) error {
+	scaledObjects := &keda.ScaledObjectList{}
+	err := r.Client.List(r.ctx, scaledObjects, client.InNamespace(instance.Namespace))
```

Comment:
if you are only interested in the presence, do a partial metadata list in current namespace with limit as 1. I suggest to look for alternatives a bit more rather than bringing this dependency.

and this api is not being registered in main, so the pr in current state will also fail for every deployment on hitting this line.